base/lock.jl

```diff
@@ -231,6 +231,24 @@ function release(s::Semaphore)
     return
 end
 
+lock(s::Semaphore) = acquire(s)
+unlock(s::Semaphore) = release(s)
+islocked(s::Semaphore) = s.curr_cnt >= s.sem_size
```

Comment:
I'm not sure if people would be suprised by this interaction:
```julia
julia> s = Base.Semaphore(2);

julia> lock(s)

julia> islocked(s)
false
```
Possibly this is why `Semaphore` isn't a subtype of `AbstractLock`?

Reply:
https://github.com/invenia/LibPQ.jl/pull/109#discussion_r292609137

Reply:
@oxinabox your thoughts?

Reply:
I would not be surprised by that interaction.
It is, after all, the key feature of a `Semaphore`